src/test/java/seedu/address/logic/commands/FilterCommandTest.java

```diff
@@ -1,13 +1,369 @@
+import seedu.address.model.employee.Leave;
+import seedu.address.model.employee.Phone;
+import seedu.address.model.employee.Role;
+import seedu.address.model.employee.Salary;
+import seedu.address.model.name.DepartmentName;
+import seedu.address.model.name.EmployeeName;
 
 public class FilterCommandTest {
+    private static final String INVALID_DEPARTMENT = "A";
+    private static final String INVALID_PHONE = "12345678";
+    private static final String INVALID_NAME = "Sam Sulek";
+    private static final String VALID_SALARY_2000 = "2000";
+    private static final String VALID_SALARY_3000 = "3000";
+    private static final String VALID_SALARY_4300 = "4300";
     private Model model = new ModelManager(getTypicalManageHr(), new UserPrefs());
     private Model expectedModel = new ModelManager(getTypicalManageHr(), new UserPrefs());
 
+    public ContainsAllPredicate createContainsAllPredicateHelper(String prefixes, String keywords) {
+        Set<EmployeeName> nameSet = new CustomSet<>();
+        Set<Phone> phoneSet = new CustomSet<>();
+        Set<Email> emailSet = new CustomSet<>();
+        Set<Address> addressSet = new CustomSet<>();
+        Set<Salary> salarySet = new CustomSet<>();
+        Set<Leave> leaveSet = new CustomSet<>();
+        Set<Role> roleSet = new CustomSet<>();
+        Set<EmployeeName> supervisorSet = new CustomSet<>();
+        Set<DepartmentName> departmentSet = new CustomSet<>();
+
+        String[] prefix = prefixes.split("_");
+        String[] keyword = keywords.split("_");
+        for (int i = 0; i < prefix.length; i++) {
+            switch (prefix[i]) {
+            case "name":
+                nameSet.add(new EmployeeName(keyword[i]));
+                break;
+            case "phone":
+                phoneSet.add(new Phone(keyword[i]));
+                break;
+            case "email":
+                emailSet.add(new Email(keyword[i]));
+                break;
+            case "address":
+                addressSet.add(new Address(keyword[i]));
+                break;
+            case "salary":
+                salarySet.add(new Salary(keyword[i]));
+                break;
+            case "leave":
+                leaveSet.add(new Leave(keyword[i]));
+                break;
+            case "role":
+                roleSet.add(new Role(keyword[i]));
+                break;
+            case "supervisor":
+                supervisorSet.add(new EmployeeName(keyword[i]));
+                break;
+            case "department":
+                departmentSet.add(new DepartmentName(keyword[i]));
+                break;
+            default:
+                break;
+            }
+        }
+
+        ContainsAllPredicate predicate = new ContainsAllPredicate(nameSet, phoneSet, emailSet, addressSet, salarySet,
+                leaveSet, roleSet, supervisorSet, departmentSet);
+        return predicate;
+    }
+
+    @Test
+    public void execute_noEmployeesFound() {
```

Comment:
Consider renaming these to the three-part format: 

> Underscores may be used in test method names using the following three part format featureUnderTest_testScenario_expectedBehavior()

as per the [Java Coding Standards](https://se-education.org/guides/conventions/java/index.html#naming)

Reply:
noted